src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+  health = 100;
 
-class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (this instanceof Carnivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0 && Animal.alive.indexOf(animal) >= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(animal), 1);
+      }
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
+}
+
+class Herbivore extends Animal {
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
```

Comment:
The `hide` method should only set the `hidden` property to `true`. If the method is intended to toggle the `hidden` property, it should be renamed to reflect that behavior (e.g., `toggleHidden`).